src/kernels/jupyter/launcher/jupyterPasswordConnect.ts

```diff
@@ -51,7 +52,15 @@ export class JupyterPasswordConnect implements IJupyterPasswordConnect {
         // See if we already have this data. Don't need to ask for a password more than once. (This can happen in remote when listing kernels)
         let result = this.savedConnectInfo.get(newUrl);
         if (!result) {
-            result = this.getNonCachedPasswordConnectionInfo(newUrl);
+            result = this.getNonCachedPasswordConnectionInfo(newUrl).then((value) => {
+                // If we fail to get a valid password connect info, don't save the value
+                traceInfo(`Password for ${newUrl} was invalid.`);
+                if (!value) {
+                    this.savedConnectInfo.delete(newUrl);
```

Comment:
This fixes the main issue. The connection info was getting saved regardless of if we validated it or now. So if you entered an incorrect password it will continue to use that for the same VS Code session without a chance to fix it.